sw/device/lib/otbn.c

```diff
@@ -0,0 +1,162 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "otbn.h"
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_otbn.h"
+#include "sw/device/lib/runtime/hart.h"
+
+static bool is_func_ptr(const otbn_t *ctx, const otbn_ptr_t *ptr) {
+  return ptr->ptr_in_cpu_addr_space >= ctx->app->imem_start &&
+         ptr->ptr_in_cpu_addr_space <= ctx->app->imem_end;
```

Comment:
This is a bit pedantic, but this is technically UB: comparisons of pointers from different allocations, or allocations that C is not aware of, are unspecified. You'll need to cast them to uintptr_t.

Reply:
uintptr_t is an optional type in C99. So now we move from "technically UB" to "hopefully support for uintptr_t", with no practical difference either way?

Reply:
uintptr_t is in every compiler we support, whereas the code you've written is absolutely UB.

Reply:
I switched to temporary variables and uintptr_t to simplify the code a bit.